### ISSUE TITLE
[P1] iPEPS excitation RDMs use interleaved physical axes as matrix axes and produce invalid traces

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#955]
> 🤖 **AI-generated issue** - written by Codex from the requested iPEPS code review.

The horizontal and vertical mixed-RDM helpers produce physical axes (ket1, bra1, ket2, bra2), then reshape directly to a matrix. The normal RDM builders explicitly transpose to (ket1, ket2, bra1, bra2) first. Consequently the excitation helpers take the wrong trace and contract Hamiltonian gates against the wrong axes. Their identity tensors also use the interleaved layout, inconsistent with the grouped Hamiltonian layout.

Sources: [horizontal reshape](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_excitations.py#L247-L251), [vertical reshape](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_excitations.py#L282-L286), [norm identity](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_excitations.py#L407-L408), [energy shift identity](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_excitations.py#L462-L464). Compare the existing transpose in [ipeps_rdm.py](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_rdm.py#L96-L103).

## Reproduction

```python
import jax.numpy as jnp
from tenax.algorithms.ipeps_config import CTMEnvironment
import tenax.algorithms.ipeps_excitations as ex

env = CTMEnvironment(
    *([jnp.ones((1, 1))] * 4 + [jnp.ones((1, 1, 1))] * 4)
)
A = jnp.array([1., 0.]).reshape(1, 1, 1, 1, 2)
B = jnp.array([0., 1.]).reshape(A.shape)
rho = ex._rdm2x1_mixed(
    A, B, env, 2, ("B", "B"), ("A", "A")
).reshape(4, 4)
print(rho)
```

Expected: `diag(0, 0, 1, 0)`, the exact density matrix of the product state |1,0>. Actual:

```text
[[0.e+00 0.e+00 0.e+00 5.e+14]
 [0.e+00 0.e+00 0.e+00 0.e+00]
 [0.e+00 0.e+00 0.e+00 0.e+00]
 [5.e+14 0.e+00 0.e+00 0.e+00]]
```

The wrong reshaping gives a zero trace; normalization then amplifies the spurious off-diagonal entries. This reproduces with an exact D=chi=1 environment.

## Suggested Fix

Use one grouped physical-axis convention throughout the mixed helpers, identity operators, and Hamiltonian contractions. Transpose interleaved outputs before matrix operations and construct the two-site identity with `eye(d*d).reshape(d,d,d,d)` in that convention. Add exact distinct-product-state tests for both orientations and agreement with standard RDMs when B=A, including complex product states. Preserve these checks when fixing the separate B-dependent-normalization defect.

Reproduced on commit `42aef1f1a94e1ca8400f6187400aa3613e3e175e` using `JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 uv run python`. No repository code was modified for these probes.
